proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+        preserveCurrent();
+        getChar();
+        if (currChar == '*') {
+          String s = currText.toString();
+          comment();
+          currText.replace(0, currText.length(), s);
+          preserveDrop();
+          continue;
+        }
+      }
+      break;
+    }
+    ProToken t = directive();
+    if (t != null)
+      return t;
+    return makeToken(PreprocessorParser.FILENAME);
+  }
+
+  ProToken directive() throws IOException {
+    LOGGER.trace("Entering directive()");
+
+    // Called by ampText, which has already gather the text for
+    // the *potential* directive.
+
+    String macroType = currText.toString().toLowerCase();
+
+    if ("&global-define".startsWith(macroType) && macroType.length() >= 4) {
+      appendToEOL();
+      // We have to do the define *before* getting next char.
+      macroDefine(PreprocessorParser.AMPGLOBALDEFINE);
+      getChar();
+      return makeToken(PreprocessorParser.AMPGLOBALDEFINE);
+    }
+    if ("&scoped-define".startsWith(macroType) && macroType.length() >= 4) {
+      appendToEOL();
+      // We have to do the define *before* getting next char.
+      macroDefine(PreprocessorParser.AMPSCOPEDDEFINE);
+      getChar();
+      return makeToken(PreprocessorParser.AMPSCOPEDDEFINE);
+    }
+
+    if ("&undefine".startsWith(macroType) && macroType.length() >= 5) {
+      // Append whitespace between &UNDEFINE and the target token
+      while (Character.isWhitespace(currChar)) {
+        append();
+        getChar();
+      }
+      // Append the target token
+      while ((!Character.isWhitespace(currChar)) && currInt != EOF_CHAR) {
+        append();
+        getChar();
+      }
+      // &UNDEFINE consumes up to *and including* the first whitespace
+      // after the token it undefines.
+      // At least that seems to be what Progress is doing.
+      if (currChar == '\r') {
+        append();
+        getChar();
+        if (currChar == '\n') {
+          append();
+          getChar();
+        }
+      } else if (currInt != EOF_CHAR) {
+        append();
+        getChar();
+      }
+      macroUndefine();
+      return makeToken(PreprocessorParser.AMPUNDEFINE);
+    }
+
+    if (macroType.equals("&analyze-suspend")) {
+      appendToEOL();
+      getChar();
+      return makeToken(PreprocessorParser.AMPANALYZESUSPEND);
+    }
+    if (macroType.equals("&analyze-resume")) {
+      appendToEOL();
+      getChar();
+      return makeToken(PreprocessorParser.AMPANALYZERESUME);
+    }
+    if (macroType.equals("&message")) {
+      appendToEOL();
+      getChar();
+      return makeToken(PreprocessorParser.AMPMESSAGE);
+    }
+
+    if (macroType.equals("&if")) {
+      return makeToken(PreprocessorParser.AMPIF);
+    }
+    if (macroType.equals("&then")) {
+      return makeToken(PreprocessorParser.AMPTHEN);
+    }
+    if (macroType.equals("&elseif")) {
+      return makeToken(PreprocessorParser.AMPELSEIF);
+    }
+    if (macroType.equals("&else")) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move the "&else" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1132)